program/src/duplicate_block_proof.rs

```diff
@@ -263,19 +405,82 @@ mod tests {
     const VERSION: u16 = 0;
 
     fn generate_proof_data<'a>(
+        leader: &'a Pubkey,
         shred1: &'a SolanaShred,
         shred2: &'a SolanaShred,
-    ) -> DuplicateBlockProofData<'a> {
-        DuplicateBlockProofData {
-            shred1: shred1.payload().as_slice(),
-            shred2: shred2.payload().as_slice(),
-        }
+    ) -> (DuplicateBlockProofData<'a>, DuplicateBlockProofContext<'a>) {
+        // Hack to simulate the merkle roots being stored in instruction data
+        let expected_shred1_merkle_root =
+            unsafe { &*Box::into_raw(Box::new(shred1.merkle_root().unwrap_or_default())) };
+        let expected_shred2_merkle_root =
+            unsafe { &*Box::into_raw(Box::new(shred2.merkle_root().unwrap_or_default())) };
```

Comment:
I know it's only testing, but this is a bit brittle, no? If the underlying `Hash` ever gets cleaned up we'll get hard-to-reproduce memory issues during testing.

I appreciate it's more annoying to use, but do you mind passing in the merkle root refs as extra parameters to the function? That way we can avoid doing this.

Reply:
yep good call I was just lazy haha ff6e4debca74efe2a318d98be24b73f2d2089911